pkg/oc/cli/cmd/newapp.go

```diff
@@ -229,6 +230,7 @@ func NewCmdNewApplication(name, baseName string, f *clientcmd.Factory, in io.Rea
 	}
 
 	cmd.Flags().BoolVar(&config.AsTestDeployment, "as-test", config.AsTestDeployment, "If true create this application as a test deployment, which validates that the deployment succeeds and then scales down.")
+	cmd.Flags().BoolVar(&config.DeployConfig, "as-deployment-config", config.DeployConfig, "If true create this application as a deployment config, which allows for hooks and custom strategies.")
```

Comment:
Why not `--as-deployment` which would turn on the k8s deployments, seems shorter and generally you enable the flag, and not disable as with the current approach.

Reply:
@soltysh the intention is to switch to use deployments by default. Adding negative flag will make the transition to new default behavior easier (we just change the default from true to false).

Reply:
Nice :) /me likes it